apps/dashboard/src/components/side-navigation/changelog-cards.tsx

```diff
@@ -0,0 +1,207 @@
+import { motion } from 'motion/react';
+import { RiCloseLine } from 'react-icons/ri';
+import { useQuery } from '@tanstack/react-query';
+import { useTelemetry } from '@/hooks/use-telemetry';
+import { TelemetryEvent } from '@/utils/telemetry';
+
+type Changelog = {
+  id: string;
+  date: string;
+  title: string;
+  notes?: string;
+  version: number;
+  imageUrl?: string;
+  published: boolean;
+};
+
+type CachedChangelogs = {
+  data: Changelog[];
+  timestamp: number;
+};
+
+const CONSTANTS = {
+  CHANGELOG_API_URL: 'https://productlane.com/api/v1/changelogs/f13f1996-c9b0-4fea-8ee7-2c3faf6a832d',
+  CACHE_KEY: 'cached_changelogs',
+  CACHE_EXPIRY: 4 * 60 * 60 * 1000, // 4 hours
+  NUMBER_OF_CARDS: 3,
+  DISMISSED_STORAGE_KEY: 'dismissed_changelogs',
+  CARD_OFFSET: 10,
+  SCALE_FACTOR: 0.06,
+} as const;
+
+export function ChangelogStack() {
+  const [changelogs, setChangelogs] = useState<Changelog[]>([]);
+  const track = useTelemetry();
+
+  const fetchChangelogs = async (): Promise<Changelog[]> => {
+    const cachedData = storage.getCachedData();
+    if (cachedData) {
+      return filterChangelogs(cachedData, storage.getDismissedChangelogs());
+    }
+
+    const response = await fetch(CONSTANTS.CHANGELOG_API_URL);
+    const rawData = await response.json();
+
+    storage.cacheData(rawData);
+
+    return filterChangelogs(rawData, storage.getDismissedChangelogs());
+  };
+
+  const { data: fetchedChangelogs } = useQuery({
+    queryKey: ['changelogs'],
+    queryFn: fetchChangelogs,
+    staleTime: CONSTANTS.CACHE_EXPIRY,
+  });
+
+  useEffect(() => {
+    if (fetchedChangelogs) {
+      setChangelogs(fetchedChangelogs);
+    }
+  }, [fetchedChangelogs]);
+
+  const handleChangelogClick = (changelog: Changelog) => {
+    track(TelemetryEvent.CHANGELOG_ITEM_CLICKED, { title: changelog.title });
+    window.open('https://roadmap.novu.co/changelog/' + changelog.id, '_blank');
+
+    storage.addToDismissed(changelog.id);
+    setChangelogs((prev) => prev.filter((log) => log.id !== changelog.id));
+  };
+
+  const handleDismiss = (e: React.MouseEvent, changelog: Changelog) => {
+    e.stopPropagation();
+    track(TelemetryEvent.CHANGELOG_ITEM_DISMISSED, { title: changelog.title });
+    storage.addToDismissed(changelog.id);
+    setChangelogs((prev) => prev.filter((log) => log.id !== changelog.id));
+  };
+
+  return (
+    <div className="absolute bottom-10 w-full">
+      <div className="m-full relative mb-4 h-[190px]">
+        {changelogs.map((changelog, index) => (
+          <ChangelogCard
+            key={changelog.id}
+            changelog={changelog}
+            index={index}
+            totalCards={changelogs.length}
+            onDismiss={handleDismiss}
+            onClick={handleChangelogClick}
+          />
+        ))}
+      </div>
+    </div>
+  );
+}
+
+const storage = {
+  getDismissedChangelogs: (): string[] => {
+    const dismissed = localStorage.getItem(CONSTANTS.DISMISSED_STORAGE_KEY);
```

Comment:
i think we started to use Clerk context for such cases maybe we can utilize it in here as well